Sanitize ingested labels

cliffcolvin opened the pull request:
## What does this PR change?
* This PR sanitizes ingested labels that are stored prefixed by Prometheus with `label_`. It also stops ingesting labels that are Prometheus labels which aren't prefixed with `label_`.

## Does this PR relate to any other PRs?
* 

## How will this PR impact users?
* Users will no longer need to prefix their queries by label with `label_` in front of their label key.
* Any old data would need to be re-ingested to have this change.

## Does this PR address any GitHub or Zendesk issues?
* Closes ...

## How was this PR tested?
* Unit test added for code coverage and accuracy.


## Does this PR require changes to documentation?
* 

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next OpenCost release? If not, why not?
*

jjarrett21 committed:
adds follow up label workflow

jjarrett21 committed:
checks for cloud cost enablement

jjarrett21 committed:
style cleanup

jjarrett21 committed:
only show when there aren't any additional report errors

jjarrett21 committed:
adds missing check

jjarrett21 committed:
order test

jjarrett21 committed:
again

jjarrett21 committed:
changes docs link and opens in new

Sean-Holcomb committed:
GCP CUD Amortization

Signed-off-by: Sean Holcomb <seanholcomb@gmail.com>

cliffcolvin committed:
sanitize ingested labels

Signed-off-by: Cliff Colvin <ccolvin@kubecost.com>

nikovacevic commented:
@cliffcolvin code looks good! Let me know when you've got testing notes, or if you need help with that, and I'll return for a definitive review.

saweber committed:
create development enviornment with tilt.dev

Signed-off-by: saweber <saweber@gmail.com>

ameijer reviewed (approved):
this LGTM. I reviewed the underlying implementation and this should accomplish exactly what we are looking for.

Cliff, it would be really easy to write a quick unit test here with some mock prom metrics just to ensure this behaves exactly the way you are looking for

kaelanspatel committed:
Remove erroneous conditional in azure ccsr parse

Signed-off-by: Kaelan Patel <kaelanspatel@gmail.com>

ameijer committed:
trying new approach for secure results uploading (#2331)

* trying new approach for secure results uploading

Signed-off-by: Alex Meijer <ameijer@kubecost.com>

* fix coverage syntax

Signed-off-by: Alex Meijer <ameijer@kubecost.com>

---------

Signed-off-by: Alex Meijer <ameijer@kubecost.com>

ameijer committed:
Re impl code coverage (#2332)

* trying new approach for secure results uploading

Signed-off-by: Alex Meijer <ameijer@kubecost.com>

* fix coverage syntax

Signed-off-by: Alex Meijer <ameijer@kubecost.com>

* fix naming, workflow trigger

Signed-off-by: Alex Meijer <ameijer@kubecost.com>

* double quote

Signed-off-by: Alex Meijer <ameijer@kubecost.com>

---------

Signed-off-by: Alex Meijer <ameijer@kubecost.com>
Signed-off-by: Alex Meijer <ameijer@users.noreply.github.com>

ameijer committed:
provide missing params (#2333)

Signed-off-by: Alex Meijer <ameijer@kubecost.com>

ameijer committed:
quotes causing mismatch (#2334)

Signed-off-by: Alex Meijer <ameijer@kubecost.com>

alrs committed:
pkg/kubecost: fix clobbered errors (#2337)

Signed-off-by: Lars Lehtonen <lars.lehtonen@gmail.com>

lmello committed:
Add Zendesk to Opencost ADOPTERS.md

We are using opencost as part of our k8s cost attribution system v2. 

The previous version of the k8s cost attribution system was built in-house and had many limitations that were improved by switching to opencost on the new version. 

We use opencost API only, we don't use opencost-ui, since we have our internal data processing pipeline and visualization system for all our cloud spending and budget tracking. 

Signed-off-by: Leonardo Rodrigues de Mello <lrodriguesdemello@zendesk.com>

AjayTripathy committed:
Update provider.go

Signed-off-by: Ajay Tripathy <ajay@kubecost.com>

mattray committed:
Updated URL and other warning messages to reflect OpenCost not Kubecost

Signed-off-by: Matt Ray <github@mattray.dev>

cliffcolvin committed:
add test for buildLabelsMap

Signed-off-by: Cliff Colvin <ccolvin@kubecost.com>

santosh-sahoo334 committed:
Fix-https://github.com/opencost/opencost/issues/2343

cliffcolvin committed:
update poolNameHelper to stop warnings

Signed-off-by: Cliff Colvin <ccolvin@kubecost.com>
